### PR TITLE
refactor: log new row groups added to read buffer

### PR DESCRIPTION
**Pull request by e-dard**
Closes https://github.com/influxdata/conductor/issues/253

This PR adds some logging to the Read Buffer to give me an idea of what the current situation is within our internal monitoring cluster.

I am emitting a fair bit of noise at `info` at the moment, which I will quieten down in due course and turn into alternative telemetry.

### CODE REVIEW

**Comment by e-dard** [comment]
Also closes #253

File: read_buffer/Cargo.toml [hunk 1]
```diff
@@ -18,6 +18,7 @@ either = "1.6.1"
 hashbrown = "0.9.1"
 internal_types = { path = "../internal_types" }
 itertools = "0.9.0"
+"observability_deps" = { path = "../observability_deps" }
```
Comment (alamb) [inline read_buffer/Cargo.toml:21]: oof - the walled garden of the read buffer with low compile times is coming to an end I suspect

File: read_buffer/src/chunk.rs [hunk 2]
```diff
@@ -236,10 +237,26 @@ impl Chunk {
...
         let table_name = table_name.into();
 
+        let rg_size = row_group.size();
+        info!(
+            "RB row group added: rb_size {:?} rg_size {:?} ({:?}%)\n{}",
+            rb_size,
+            rg_size,
+            rg_size as f64 / rb_size as f64,
+            &row_group,
+        );
```
Comment (alamb) [inline read_buffer/src/chunk.rs:258]: If you want, you can emit `rb_size` etc as structured fields (and get the `rg_size=...` for free with something like 

```suggestion
        let rg_size = row_group.size();
        let percent =  rg_size as f64 / rb_size as f64;
        info!(
            rb_size, rg_size, percent,
            ?row_group,
            "RB row group added",
        );
```

I also don't really sure how helpfl the

**Review by alamb — APPROVED** [review]
LGTM

File: read_buffer/src/chunk.rs (hunk 2 above)
Comment (domodwyer) [reply read_buffer/src/chunk.rs]: +1

Keeping the log messages fairly static but having the data as structured fields is super invaluable for retrospective debugging via log analysis - it saved my skin many times after a customer dumped a load of logs on us post-incident.

File: read_buffer/src/chunk.rs [hunk 3]
```diff
@@ -236,10 +237,21 @@ impl Chunk {
...
+            .map(|c| c.get_buffer_memory_size())
+            .sum::<usize>();
+
         // This call is expensive. Complete it before locking.
         let row_group = RowGroup::from(table_data);
         let table_name = table_name.into();
 
+        let rg_size = row_group.size();
+        let compression = format!("{:.2}%", (rg_size as f64 / rb_size as f64) * 100.0);
+        info!(rb_size, rg_size, %compression, %row_group, "row group added");
```
Comment (domodwyer) [inline read_buffer/src/chunk.rs:253]: Nice - I'm excited to see these numbers! 🚀

**Review by domodwyer — APPROVED** [review]

**Pull request merged**